test/Microsoft.IdentityModel.Tokens.Extensions.Tests/KeyVaultKeyWrapProviderTests.cs

```diff
@@ -0,0 +1,151 @@
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Azure.KeyVault;
+using Microsoft.IdentityModel.Tests;
+using Xunit;
+
+#pragma warning disable CS3016 // Arrays as attribute arguments is not CLS-compliant
+
+namespace Microsoft.IdentityModel.Tokens.Extensions.Tests
+{
+    public class KeyVaultKeyWrapProviderTests
+    {
+        private readonly IKeyVaultClient _client;
+        private readonly SecurityKey _key;
+
+        public KeyVaultKeyWrapProviderTests()
+        {
+            _client = new MockKeyVaultClient();
+            _key = new KeyVaultSecurityKey(KeyVaultUtilities.CreateKeyIdentifier(), keySize: default);
+        }
+
+        public static TheoryData<KeyWrapProviderTheoryData> DisposeProviderTheoryData
+        {
+            get => new TheoryData<KeyWrapProviderTheoryData>
+            {
+                new KeyWrapProviderTheoryData
+                {
+                    Algorithm = SecurityAlgorithms.RsaPKCS1,
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = nameof(SecurityAlgorithms.RsaPKCS1),
+                },
+                new KeyWrapProviderTheoryData
+                {
+                    Algorithm = SecurityAlgorithms.RsaOAEP,
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = nameof(SecurityAlgorithms.RsaOAEP),
+                },
+            };
+        }
+
+        public static TheoryData<KeyWrapProviderTheoryData> KeyWrapProviderTheoryData
+        {
+            get => new TheoryData<KeyWrapProviderTheoryData>
+            {
+                new KeyWrapProviderTheoryData
+                {
+                    Algorithm = null,
+                    ExpectedException = ExpectedException.ArgumentNullException(),
+                    First = true,
+                    TestId = "NullAlgorithm",
+                },
+                new KeyWrapProviderTheoryData
+                {
+                    Algorithm = string.Empty,
+                    ExpectedException = ExpectedException.ArgumentNullException(),
+                    TestId = "EmptyAlgorithm",
+                },
+                new KeyWrapProviderTheoryData
+                {
+                    Algorithm = SecurityAlgorithms.RsaPKCS1,
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = nameof(SecurityAlgorithms.RsaPKCS1),
+                },
+                new KeyWrapProviderTheoryData
+                {
+                    Algorithm = SecurityAlgorithms.RsaOAEP,
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = nameof(SecurityAlgorithms.RsaOAEP),
+                },
+            };
+        }
+
+        [Theory, MemberData(nameof(DisposeProviderTheoryData))]
```

Comment:
I believe in most of our tests we put the TheoryData directly after the test method that is using it.

Reply:
Fixed: Moved the TheoryData directly after the test methods that are using them.